transport/wsn/src/test/java/org/kaazing/gateway/transport/wsn/specification/ws/connector/FragmentationIT.java

```diff
@@ -0,0 +1,1029 @@
+                .withLookingForStuckThread(true).build());
+    private final TestRule trace = new MethodExecutionTrace();
+
+    @Rule
+    public TestRule chain = RuleChain.outerRule(trace).around(timeoutRule).around(k3po).around(connector);
+
+    private static String TEXT_FILTER_NAME = WsnProtocol.NAME + "#text";
+
+    private Mockery context;
+
+    @Before
+    public void initialize() {
+        context = new Mockery() {
+            {
+                setImposteriser(ClassImposteriser.INSTANCE);
+            }
+        };
+        context.setThreadingPolicy(new Synchroniser());
+    }
+
+    @Test
+    @Ignore("IllegalArgumentException: message is empty. Forgot to call flip")
+    @Specification({
+        "server.echo.binary.payload.length.0.fragmented.with.injected.ping.pong/handshake.response.and.frames" })
+    public void shouldEchoServerSendBinaryFrameWithEmptyPayloadFragmentedAndInjectedPingPong() throws Exception {
+        final IoHandler handler = context.mock(IoHandler.class);
+        final List<ByteBuffer> bufferList = new ArrayList<ByteBuffer>();
+
+        context.checking(new Expectations() {
+            {
+                oneOf(handler).sessionCreated(with(any(IoSessionEx.class)));
+                oneOf(handler).sessionOpened(with(any(IoSessionEx.class)));
+                allowing(handler).messageReceived(with(any(IoSessionEx.class)), with(any(Object.class)));
+                will(new CustomAction("Capture messageReceived() parameters") {
+                    @Override
+                    public Object invoke(Invocation invocation) throws Throwable {
+                        Object message = invocation.getParameter(1);
+                        IoBufferEx ioBuffer = (IoBufferEx) message;
+                        bufferList.add(ioBuffer.buf());
+
+                        WsnSession wsnConnectSession = (WsnSession) invocation.getParameter(0);
+                        if (wsnConnectSession != null) {
+                            IoFilterChain filterChain = wsnConnectSession.getFilterChain();
+                            IoBufferAllocatorEx<? extends WsBuffer> allocator = wsnConnectSession.getBufferAllocator();
+
+                            final boolean hasPostUpgradeChildWsnSession = wsnConnectSession.getHandler() == handler;
+                            final ResourceAddress wsnSessionLocalAddress = wsnConnectSession.getLocalAddress();
+                            final boolean isLightweightWsnSession = wsnSessionLocalAddress.getOption(LIGHTWEIGHT);
+                            boolean sendMessagesDirect = isLightweightWsnSession
+                                                         && hasPostUpgradeChildWsnSession; // post-upgrade
+
+                            if (sendMessagesDirect) {
+                                filterChain.fireMessageReceived(message);
+                                return null;
+                            }
```

Comment:
This code (line 88 through 97) is not needed. We are not doing wsx here, it's straight wsn.